### PR TITLE
Add recommendation for priorityClass

### PR DESCRIPTION
**Pull request by kfox1111**

### CODE REVIEW

**Comment by faisal-memon** [comment]
@kfox1111 Thanks for submitting this. Where are the actual priority class values being set?

**Comment by kfox1111** [comment]
> Where are the actual priority class values being set?

I think your asking for this?
https://github.com/spiffe/helm-charts-hardened/pull/124/files#diff-789477a4f2a924ca09b7f80a469fc8827159a3d3d8a6281c37018dc16b60c094

**Comment by faisal-memon** [comment]
> > Where are the actual priority class values being set?
> 
> I think your asking for this? https://github.com/spiffe/helm-charts-hardened/pull/124/files#diff-789477a4f2a924ca09b7f80a469fc8827159a3d3d8a6281c37018dc16b60c094

Got it, its just the single value. I thought it was a block.

File: charts/spire/values.yaml [hunk 1]
```diff
@@ -22,8 +22,10 @@ global:
     upstreamServerAddress: ""
 
     ## @param global.spire.recommendations.enabled Use recommended settings for production deployments. Default is off.
+    ## @param global.spire.recommendations.priorityClasses Set to true to use our recommendations for pod priority classes
     recommendations:
       enabled: false
+      priorityClasses: true
```
Comment (faisal-memon) [inline charts/spire/values.yaml:28]:
```suggestion
      priorityClasseName: true
```

File: charts/spire/values.yaml [hunk 2]
```diff
@@ -22,8 +22,10 @@ global:
     upstreamServerAddress: ""
 
     ## @param global.spire.recommendations.enabled Use recommended settings for production deployments. Default is off.
+    ## @param global.spire.recommendations.priorityClasses Set to true to use our recommendations for pod priority classes
```
Comment (faisal-memon) [inline charts/spire/values.yaml:25]:
```suggestion
    ## @param global.spire.recommendations.priorityClasses Set to true to use our recommendations for Pod Priority Class Names
```

File: charts/spire/templates/_spire-lib.tpl [hunk 3]
```diff
@@ -247,3 +247,19 @@ to merge in values, but spire needs arrays.
 {{- $_ := set $config "plugins" $plugins }}
 {{- $config | toPrettyJson }}
 {{- end }}
+
+{{- define "spire-lib.default_node_priority_class_name" }}
+{{- if .Values.priorityClassName }}
+priorityClassName: {{ .Values.priorityClassName }}
+{{- else if and (dig "spire" "recommendations" "enabled" false .Values.global) (dig "spire" "recommendations" "priorityClasses" true .Values.global) }}
```
Comment (faisal-memon) [inline charts/spire/templates/_spire-lib.tpl:254]:
```suggestion
{{- else if and (dig "spire" "recommendations" "enabled" false .Values.global) (dig "spire" "recommendations" "priorityClassName" true .Values.global) }}
```

File: charts/spire/templates/_spire-lib.tpl [hunk 4]
```diff
@@ -247,3 +247,19 @@ to merge in values, but spire needs arrays.
...
+priorityClassName: system-node-critical
+{{- end }}
+{{- end }}
+
+{{- define "spire-lib.default_cluster_priority_class_name" }}
+{{- if .Values.priorityClassName }}
+priorityClassName: {{ .Values.priorityClassName }}
+{{- else if and (dig "spire" "recommendations" "enabled" false .Values.global) (dig "spire" "recommendations" "priorityClasses" true .Values.global) }}
```
Comment (faisal-memon) [inline charts/spire/templates/_spire-lib.tpl:262]:
```suggestion
{{- else if and (dig "spire" "recommendations" "enabled" false .Values.global) (dig "spire" "recommendations" "priorityClassName" true .Values.global) }}
```

File: charts/spire/values.yaml [hunk 5]
```diff
@@ -22,8 +22,10 @@ global:
     upstreamServerAddress: ""
 
     ## @param global.spire.recommendations.enabled Use recommended settings for production deployments. Default is off.
+    ## @param global.spire.recommendations.priorityClassName Set to true to use our recommendations for Pod Priority Class Names
```
Comment (faisal-memon) [inline charts/spire/values.yaml:25]:
```suggestion
    ## @param global.spire.recommendations.priorityClassName Set to true to use recommended values for Pod Priority Class Names
```

**Review by faisal-memon — APPROVED** [review]

**Pull request merged**